detekt-rules-naming/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/BooleanPropertyNaming.kt

```diff
@@ -0,0 +1,101 @@
+package io.gitlab.arturbosch.detekt.rules.naming
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.LazyRegex
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.api.internal.Configuration
+import io.gitlab.arturbosch.detekt.api.internal.RequiresTypeResolution
+import io.gitlab.arturbosch.detekt.rules.fqNameOrNull
+import io.gitlab.arturbosch.detekt.rules.identifierName
+import org.jetbrains.kotlin.psi.KtCallableDeclaration
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.typeBinding.createTypeBindingForReturnType
+
+/**
+ * Reports when a boolean property doesn't have one of the following prefixes:
+ * is|has|are.
+ *
+ * <noncompliant>
+ * val progressBar: Boolean = true
+ * </noncompliant>
+ *
+ * <compliant>
+ * val hasProgressBar: Boolean = true
+ * </compliant>
+ */
+@RequiresTypeResolution
+class BooleanPropertyNaming(config: Config = Config.empty) : Rule(config) {
+
+    @Configuration("naming pattern")
+    private val allowedPattern by LazyRegex(
+        ALLOWED_PREFIXES,
+        "^(is|has|are)"
+    )
```

Comment:
```suggestion
    private val allowedPattern: Regex by config("^(is|has|are)", String::toRegex)
```
`LazyRegex` will be marked deprecated soon.

Reply:
I think this confusing to the user. We should either allow users to specify regular expression patterns that match the entire boolean property or have a list of prefixes. Having a pattern that matches only parts of the name to me feels rather unusual.